pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1068,23 +1061,32 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
                                         });
                             }
                             return loadOrCreatePersistentTopic(tpName, createIfMissing, properties, topicPolicies);
-                        }).thenCompose(optionalTopic -> {
-                            if (!optionalTopic.isPresent() && createIfMissing) {
-                                log.warn("[{}] Try to recreate the topic with createIfMissing=true "
-                                        + "but the returned topic is empty", topicName);
-                                return getTopic(topicName, createIfMissing, properties);
+                        });
+                        topicFuture.whenComplete((__, ex) -> {
```

Comment:
If the topic's loading is slow enough, this line may cause an OOM error because too many `when complete` were appended.